assets/search/components/TopicList.tsx

```diff
@@ -63,21 +79,103 @@ const TopicList = ({
                 topics={filteredTopics}
                 folderPopover={folderPopover}
                 toggleFolderPopover={toggleFolderPopover}
-                moveTopic={moveTopic}
                 saveFolder={saveFolder}
                 deleteFolder={deleteFolder}
+                opened={openedFolders[folder._id] ?? false}
+                setOpened={(val) => {
+                    setOpenedFolders({
+                        ...openedFolders,
+                        [folder._id]: val,
+                    });
+                }}
             >
-                {filteredTopics.map(renderTopic)}
+                {filteredTopics.map((topic) => renderTopic(topic))}
             </TopicFolder>
         );
     });
 
-    const renderedTopics = topics.filter((topic: any) => topic.folder == null).map(renderTopic);
+    const topicsWithoutFolder = topics.filter((topic) => topic.folder == null).map((topic) => renderTopic(topic));
+
+    const sensors = useCustomSensors({activationConstraint: {distance: 10}});
 
     return (
         <>
-            {renderedFolders}
-            {renderedTopics}
+            <DndContext
```

Comment:
I wonder if it would be possible to have a reusable component for the context and the overlay which would get the children and `onDragEnd` handler

Reply:
Restoring scroll was the ugly repetetive part IMO, I've moved it into a custom hook. I think not making that component is cleaner for now. We can come back when we have more use cases and see if it would help to abstract it further.